src/engine/source/kvdb/src/refCounter.cpp

```diff
@@ -0,0 +1,58 @@
+#include <kvdb/refCounter.hpp>
+
+namespace kvdbManager
+{
+
+RefCounter::RefCounter(const RefCounter& other)
+{
+    m_refMap = other.m_refMap;
+}
+
+void RefCounter::addRef(const std::string& name, const unsigned int times)
+{
+    m_refMap[name] += times;
+}
+
+void RefCounter::removeRef(const std::string& name)
+{
+    if (m_refMap.count(name) > 0)
+    {
+        m_refMap[name]--;
+        if (0 == m_refMap[name])
+        {
+            m_refMap.erase(name);
+        }
+    }
+}
+
+unsigned int RefCounter::count(const std::string& name) const
+{
+    if (m_refMap.count(name) > 0)
+    {
+        return m_refMap.at(name);
+    }
```

Comment:
Should not do the search twice, nor should a previous check be done so as not to launch an execption:

```suggestion
    auto it = m_refMap.find(name);
    if (it != m_refMap.end())
    {
        return it->second;
    }
```